scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkCounters.scala

```diff
@@ -0,0 +1,104 @@
+package com.twitter.scalding.spark_backend
+
+import com.twitter.scalding.{ExecutionCounters, StatKey}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.util.AccumulatorV2
+
+import scala.collection.mutable.{Map => MutableMap}
+
+/**
+ * Spark Counters is a singleton responsible for managing counters in the spark backend
+ *
+ * Inspiration has been taken from
+ * https://stackoverflow.com/questions/66438570/create-accumulator-on-executor-dynamically
+ */
+object SparkCounters {
+
+  var accumulator: SparkCountersInternal = null
```

Comment:
1. can we make this private?
2. I would probably instead do `@transient private lazy val accumulator = new AtomicReference[SparkCountersInternal](null)` and use atomic operations since I think you can have multithreading in spark.

Reply:
I don't think 2 is necessary. Check out this explanation by someone familiar with spark internals: 
https://stackoverflow.com/a/43912747

> Accumulators are not thread-safe, in fact they do not need to be thread-safe. For executors, accumulators are write only variables, they can be added by executors and they can be read by the driver. Driver makes use of DAGScheduler.updateAccumulators method to update values of accumulators after task is completed, and this method is called only from a thread that runs scheduling loop. At a time, only one task completion event is handled. That's why there is no need for accumulators to be thread-safe.

Reply:
for 1 I changed the implementation so not sure if it's still applicable